src/main/kotlin/interactor/GetCostlierCityWithRentingAnApartmentBetweenCityCenterAndOutsideCityCenterInteractor.kt

```diff
@@ -18,29 +17,37 @@ class GetCostlierCityWithRentingAnApartmentBetweenCityCenterAndOutsideCityCenter
     }
 
     private fun getThreeBedroom(): CityEntity {
-
         return dataSource.getAllCitiesData()
-            .filter {
-                it.dataQuality
-                        && it.realEstatesPrices.apartment3BedroomsInCityCentre != null
-                        && it.realEstatesPrices.apartment3BedroomsOutsideOfCentre != null
-            }
-            .maxByOrNull { abs(it.realEstatesPrices.apartment3BedroomsInCityCentre!! - it.realEstatesPrices.apartment3BedroomsOutsideOfCentre!!) }!!
-
+            .filter(::excludeNullPriceApartment3BedroomsAndLowQualityData)
+            .maxByOrNull(::getDifferencePricesApartment3Bedrooms)!!
     }
 
     private fun getOneBedroom(): CityEntity {
         return dataSource.getAllCitiesData()
-            .filter {
-                it.dataQuality
-                        && it.realEstatesPrices.apartmentOneBedroomInCityCentre != null
-                        && it.realEstatesPrices.apartmentOneBedroomOutsideOfCentre != null
-            }
+            .filter(::excludeNullPriceApartment1BedroomsAndLowQualityData)
+            .maxByOrNull(::getDifferencePricesApartment1Bedrooms)!!
+    }
 
-            .maxByOrNull { abs(it.realEstatesPrices.apartmentOneBedroomInCityCentre!! - it.realEstatesPrices.apartmentOneBedroomOutsideOfCentre!!) }!!
 
+    private fun excludeNullPriceApartment3BedroomsAndLowQualityData(city: CityEntity): Boolean {
+        return city.dataQuality
+                && city.realEstatesPrices.apartment3BedroomsInCityCentre != null
+                && city.realEstatesPrices.apartment3BedroomsOutsideOfCentre != null
+    }
 
+    private fun excludeNullPriceApartment1BedroomsAndLowQualityData(city: CityEntity): Boolean {
+        return city.dataQuality
+                && city.realEstatesPrices.apartmentOneBedroomInCityCentre != null
+                && city.realEstatesPrices.apartmentOneBedroomOutsideOfCentre != null
     }
 
 
+    private fun getDifferencePricesApartment3Bedrooms(city: CityEntity): Float {
+        return abs(city.realEstatesPrices.apartment3BedroomsInCityCentre!! - city.realEstatesPrices.apartment3BedroomsOutsideOfCentre!!)
+    }
+
+    private fun getDifferencePricesApartment1Bedrooms(city: CityEntity): Float {
+        return abs(city.realEstatesPrices.apartmentOneBedroomInCityCentre!! - city.realEstatesPrices.apartmentOneBedroomOutsideOfCentre!!)
+    }
+
 }
```

Comment:
I don't know how, but the repetition of the two cases — (3 bedrooms and 1 bedroom) with the same logic could be generalized.